File: java/parser/src/main/java/gov/lbl/parser/parser/Enum_listVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Enum_list;
import gov.lbl.parser.domain.Enumeration_literal;
import static java.util.stream.Collectors.toList;

import java.util.List;

public class Enum_listVisitor extends modelicaBaseVisitor<Enum_list> {
    @Override
    public Enum_list visitEnum_list(modelicaParser.Enum_listContext ctx) {
        Enumeration_literalVisitor enumeration_literalVisitor = new Enumeration_literalVisitor();
        List<Enumeration_literal> enumeration_literal_list = ctx.enumeration_literal().stream().map(enumeration_literal -> enumeration_literalVisitor.visitEnumeration_literal(enumeration_literal)).collect(toList());
        return new Enum_list(enumeration_literal_list);
    }
}

File: jsParser/parser/Enum_listVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor;
const Enum_list = require('../domain/Enum_list');

const Enumeration_literalVisitor = require('./Enumeration_literalVisitor');

class Enum_listVisitor {
    constructor() {
        modelicaVisitor.call(this);
        return this;
    }
    visitEnum_list(ctx) {
        var enum_list = []

        if (ctx.enumeration_literal()) {
            var enumeration_literalVisitor = Enumeration_literalVisitor.Enumeration_literalVisitor();
            ctx.enumeration_literal().forEach(enumeration_literal => {
                enum_list.push(enumeration_literalVisitor.visitEnumeration_literal(enumeration_literal));
            });
        }

        return new Enum_list.Enum_list(enum_list);
    }
};

Enum_listVisitor.prototype = Object.create(modelicaVisitor.prototype);

exports.visitEnum_list = this.visitEnum_list;
exports.Enum_listVisitor = Enum_listVisitor;